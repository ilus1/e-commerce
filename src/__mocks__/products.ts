export const mockProducts = [
  { id: '1', name: 'Produto 1', price: 10, photo: 'url1', description: 'Descrição 1' },
];

export const mockCartItems = [
  { id: '1', name: 'Produto 1', price: 10, photo: 'url1', description: 'Descrição 1', count: 1 },
];
