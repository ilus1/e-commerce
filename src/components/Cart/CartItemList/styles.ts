import styled from "styled-components";

export const Container = styled.ul`
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  gap: 28px;
  padding: 20px 30px;
  margin: 0;
  overflow-y: auto;
  overflow-x: hidden;
`;