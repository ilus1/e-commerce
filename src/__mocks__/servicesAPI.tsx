import { rest } from "msw";
import { setupServer } from "msw/node";

import { mockProducts } from "./products";

const server = setupServer(
  rest.get('*', (_, res, ctx) => res(ctx.status(200), ctx.json(mockProducts)))
);

export default server;
