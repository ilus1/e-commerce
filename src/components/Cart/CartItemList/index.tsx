import CartItemComponent from '../CartItem'
import { Container } from './styles';

interface CartItem {
  id: string
  name: string
  price: number
  photo: string
  description: string
  count: number
}

interface CartItemListProps {
  cartItemList: CartItem[]
  onRemoveItem: (cartItem: CartItem) => void
  onRemoveProduct: (cartItem: CartItem) => void
  onAddItem: (cartItem: CartItem) => void
}

function CartItemList({ cartItemList, onRemoveItem, onAddItem, onRemoveProduct }: CartItemListProps) {
  return (
    <Container>
      {cartItemList.map((cartItem) => (
        <CartItemComponent
          key={cartItem.id}
          onRemove={onRemoveProduct}
          item={cartItem}
          onAddItem={onAddItem}
          onRemoveItem={onRemoveItem}
        />
      ))}
    </Container>
  );
}

export default CartItemList;
