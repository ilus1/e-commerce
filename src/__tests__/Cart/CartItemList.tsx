import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';

import CartItemList from '../../components/Cart/CartItemList';

import { mockCartItems } from '../../__mocks__/products';

describe('Componente da lista de itens do carrinho', () => {

  it('Testa carregamento correto do componente', () => {
    const tree = renderer
      .create(
        <CartItemList
          cartItemList={mockCartItems}
          onRemoveItem={() => {}}
          onAddItem={() => {}}
          onRemoveProduct={() => {}}
        />
      );

    expect(tree).toMatchSnapshot();
  });

  it('Testa se todos os itens do carrinho estÃ£o sendo renderizados', () => {
    const { getAllByTestId } = render(
      <CartItemList
        cartItemList={mockCartItems}
        onRemoveItem={() => {}}
        onAddItem={() => {}}
        onRemoveProduct={() => {}}
      />
    );

    const cartItemComponents = getAllByTestId('cart-item-component');
    expect(cartItemComponents.length).toBe(mockCartItems.length);
  });
});
