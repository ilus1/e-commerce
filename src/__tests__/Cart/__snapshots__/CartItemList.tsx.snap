// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente da lista de itens do carrinho Testa carregamento correto do componente 1`] = `
<ul
  className="sc-fubDmA lihiSQ"
>
  <li
    className="sc-hKgKIp fFLrTl"
    data-testid="cart-item-component"
  >
    <button
      className="sc-eCstZk bWQkgF"
      onClick={[Function]}
    >
      X
    </button>
    <img
      className="sc-gKsecS hntZsr"
      src="url1"
    />
    <span
      className="sc-jSguLX dDFcMB"
    >
      Produto 1
    </span>
    <div
      className="sc-bdfCDU fhrPBJ"
    >
      <button
        className="sc-dlfmHC jhxSjB"
        onClick={[Function]}
      >
        -
      </button>
      <p
        className="sc-gsTDqH iORjDB"
      >
        1
      </p>
      <button
        className="sc-dlfmHC jhxSjB"
        onClick={[Function]}
      >
        +
      </button>
    </div>
    <span
      className="sc-iBPTVF jogATO"
    >
      R$ 
      10
    </span>
  </li>
</ul>
`;
